src/include/migraphx/op/reduce_op.hpp

```diff
@@ -0,0 +1,186 @@
+#ifndef MIGRAPHX_GUARD_OPERATORS_OP_HPP
+#define MIGRAPHX_GUARD_OPERATORS_OP_HPP
+
+//#include <migraphx/half.hpp>
+#include <migraphx/op/name.hpp>
+#include <migraphx/check_shapes.hpp>
+#include <migraphx/argument.hpp>
+#include <migraphx/shape_for_each.hpp>
+#include <migraphx/par_for.hpp>
+#include <migraphx/config.hpp>
+#include <vector>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace op {
+
+struct lowest
+{
+    template <class T>
+    operator T() const
+    {
+        return std::numeric_limits<T>::lowest();
+    }
+};
+
+struct highest
+{
+    template <class T>
+    operator T() const
+    {
+        return std::numeric_limits<T>::max();
+    }
+};
+
+struct zero
+{
+    template <class T>
+    operator T() const
+    {
+        return T{0};
+    }
+};
+
+struct sum_op
+{
+    std::size_t elem_num = 1;
+    template <class T>
+    T operator()(T x, T y) const
+    {
+        return x + y;
+    }
+
+    auto init() const { return zero(); }
+
+    auto output() const
+    {
+        return [=](auto val) { return val; };
+    }
```

Comment:
These operators should be defined in the derived classes. We also need `input()` method as well so we can implement sum of squares.

Reply:
fixed